src/bench/bench.ts

```diff
@@ -44,12 +49,15 @@ function spawnOnce(args: { supervise?: boolean; filename: string; swc: boolean }
   }
 
   const binPath = path.resolve("pkg/esbuild-dev.bin.js");
-  const allArgs = [...extraArgs, "-r", path.resolve("pkg/bench/bench-child-hooks.js"), args.filename];
+  const root = findRoot(args.filename);
+  const relativeFilePath = path.relative(root, args.filename);
+  const allArgs = [...extraArgs, "-r", path.resolve("pkg/bench/bench-child-hooks.js"), relativeFilePath];
 
   log.debug(binPath, ...allArgs);
 
   return spawn(binPath, allArgs, {
     stdio: ["ignore", "pipe", "ignore"],
+    cwd: root,
```

Comment:
`esbuild-dev`'s main process uses [`process.cwd()`](https://github.com/gadget-inc/esbuild-dev/blob/90d34966bdf6ccb94b3e4ba1c55df9526505ca59/src/index.ts#L139) to determine workspace root, so chdir-ing into the correct root is necessary for benchmarking actual projects.